src/scripts/main.js

```diff
@@ -1,3 +1,234 @@
+
+    return false;
+  }
+
+  if (+data.get('age') < 18 || +data.get('age') > 90) {
+    pushNotification(
+      'Error',
+      'Age should be older then 18 and yonger then 99',
+      'error',
+    );
+
+    return false;
+  }
+
+  if (data.get('position').trim().length <= 0) {
+    pushNotification('Error', 'Please enter position', 'error');
+
+    return false;
+  }
+
+  if (+data.get('salary') <= 0) {
+    pushNotification('Error', 'Please enter valid salary', 'error');
+
+    return false;
+  }
+
+  return true;
+}
+
+function pushNotification(title, description, type = '') {
+  const message = document.createElement('div');
+
+  message.className = `notification ${type}`;
+  message.dataset.qa = 'notification';
+  message.style.top = `${500}px`;
+  message.style.right = `${10}px`;
+
+  message.innerHTML = `
+      <h2 class="title">${title}</h2>
+      <p>${description}</p>
+    `;
+  document.body.append(message);
+
+  setTimeout(() => {
+    message.remove();
+  }, 2000);
+}
+
+const tableRows = [...document.querySelector('tbody').rows];
+const tHead = document.querySelector('thead');
+const tBody = document.querySelector('tbody');
+const headerCells = tHead?.rows[0].cells;
+const className = 'acdSorted';
+const sortMapIndx = {};
+const formHTML = `<form
+      id="form-new-empl"
+      class="new-employee-form"
+    >
+      <label>Name:
+        <input
+          name="name"
+          type="text"
+          data-qa="name"
+        >
+      </label>
+
+      <label>Position:
+        <input
+          name="position"
+          type="text"
+          data-qa="position"
+        >
+      </label>
+
+      <label>Office:
+        <select
+          name="office"
+          type="text"
+          data-qa="office"
+        >
+          <option value="Tokyo">Tokyo</option>
+          <option value="Singapore">Singapore</option>
+          <option value="London">London</option>
+          <option value="New York">New York</option>
+          <option value="Edinburgh">Edinburgh</option>
+          <option value="San Francisco">San Francisco</option>
+          <option value="London">London</option>
+        </select>
+      </label>
+
+      <label>Age:
+        <input
+          name="age"
+          type="number"
+          data-qa="age"
+        >
+      </label>
+
+      <label>Salary:
+        <input
+          name="salary"
+          type="number"
+          data-qa="salary"
+        >
+      </label>
+
+      <button type="submit">Save to table</button>
+    </form>`;
+
+document.querySelector('table').insertAdjacentHTML('afterend', formHTML);
+
+const formElem = document.body.querySelector('#form-new-empl');
+
+for (let i = 0; i < tHead?.rows[0].cells.length; i++) {
+  sortMapIndx[headerCells[i].textContent] = i;
+}
+
+tHead?.addEventListener('click', (e) => {
+  const acdStat = e.target.classList.contains(className);
+
+  sortRows(tableRows, sortMapIndx[e.target.textContent], !acdStat);
```

Comment:
It seems like the newly added rows are not taken into account during sorting (always on the top)
You can query all rows again inside this function to make sure you use all rows